include/rviz_visual_tools/rviz_visual_tools.h

```diff
@@ -1046,7 +1058,7 @@ a   *        Warning: when using this in a loop be sure to call trigger() at end
   ros::NodeHandle nh_;
 
   // Short name for this class
-  static const std::string name_;
+  static RVIZ_VISUAL_TOOLS_DECL const std::string name_;
```

Comment:
why does a string need to be so complicated? what does this do?

Reply:
To MSVC, everything in a shared library, which will be consumed later by other modules, are required to decorated `__declspec(dllexport)`. Otherwise, those symbols won't be visible from the library you are linking against. It is something different compared to GNU GCC\CLANG, which seems to export everything by default.

To make Windows\MSVC porting easier, we add [`CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS`](https://cmake.org/cmake/help/latest/variable/CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.html) by default to catkin when it is building with MSVC. This feature generally automatically exports every functions from the shared library but doesn't fix up the thing exported from data section. So there are still [some cases](https://blog.kitware.com/create-dlls-on-windows-without-declspec-using-new-cmake-export-all-feature/) where dllexport/dllimport needs to be added by hands.

That's why only the static variables in this project get this particular fix-up.